lib/util/platform.js

```diff
@@ -318,6 +318,74 @@ shaka.util.Platform = class {
     const Platform = shaka.util.Platform;
     return Platform.isTizen() || Platform.isXboxOne();
   }
+
+  /**
+   * Returns true if we need to polyfill Apple legacy MediaKeys
+   * implementation
+   *
+   * @return {boolean}
+   */
+  static shouldPolyfillMediaKeysApple() {
+    if (!window.HTMLVideoElement || !window.WebKitMediaKeys) {
+      return false;
+    }
+
+    const safariVersion = shaka.util.Platform.safariVersion();
+
+    if (navigator.requestMediaKeySystemAccess &&
+        // eslint-disable-next-line no-restricted-syntax
+        MediaKeySystemAccess.prototype.getConfiguration &&
+        safariVersion && safariVersion >= 14) {
+      // Unprefixed EME is preferable.
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Returns true if we need to polyfill Webkit legacy MediaKeys
+   * implementation
+   *
+   * @return {boolean}
+   */
+  static shouldPolyfillMediaKeysWebkit() {
+    if (!window.HTMLVideoElement ||
+        (navigator.requestMediaKeySystemAccess &&
+          // eslint-disable-next-line no-restricted-syntax
+          MediaKeySystemAccess.prototype.getConfiguration)) {
+      return false;
+    }
+
+    // eslint-disable-next-line no-restricted-syntax
+    return !!(HTMLMediaElement.prototype.webkitGenerateKeyRequest ||
+      // eslint-disable-next-line no-restricted-syntax
+      HTMLMediaElement.prototype.generateKeyRequest);
+  }
+
+  /**
+   * Returns true if we need to polyfill MS legacy MediaKeys implementation
+   *
+   * @return {boolean}
+   */
+  static shouldPolyfillMediaKeysMS() {
+    return window.HTMLVideoElement && window.MSMediaKeys &&
+      (!navigator.requestMediaKeySystemAccess ||
+       // eslint-disable-next-line no-restricted-syntax
+       !MediaKeySystemAccess.prototype.getConfiguration);
+  }
+
+  /**
+   * Returns true if any MediaKeys polyfill should have been
+   * installed
+   *
+   * @return {boolean}
+   */
+  static isMediaKeysPolyfilled() {
```

Comment:
Maybe I'm confused here, but the JSDoc and the method name seem to be at odds.
The JSDoc suggests that this returns true if MediaKeys need to be polyfilled but haven't yet, which is also what the code looks like it is doing. The method name sounds like it should be returning true if the polyfills have already been installed.
Perhaps the name `willMediaKeysBePolyfilled` is better?